task/pnc-prebuild-git-clone-oci-ta/0.1/patch.yaml

```diff
@@ -0,0 +1,76 @@
+- op: replace
+  path: /metadata/name
+  value: pnc-prebuild-git-clone-oci-ta
+- op: replace
+  path: /metadata/annotations/tekton.dev~1displayName
+  value: pnc prebuild git clone oci trusted artifacts
+- op: replace
+  path: /spec/description
+  value: The pnc-prebuild-git-clone-oci-ta Task will clone a repo from the provided url, apply
+    PNC prebuild modifications and store it as a trusted artifact in the provided OCI repository.
+
+- op: add
+  path: /spec/params/-
+  value:
+    name: BUILD_SCRIPT
+    description: The build script to embed with the Containerfile
+    type: string
+- op: add
+  path: /spec/params/-
+  value:
+    name: BUILD_TOOL
+    description: The build tool to use (ant, gradle, maven, sbt).
+    type: string
+- op: add
+  path: /spec/params/-
+  value:
+    name: BUILD_TOOL_VERSION
+    description: The build tool version to use (e.g. 3.9.5)
+    type: string
+- op: add
+  path: /spec/params/-
+  value:
+    name: JAVA_VERSION
+    description: Java version to use (7, 8, 9, 11, 17, 21, 22, 23)
+    type: string
+- op: add
+  path: /spec/params/-
+  value:
+    name: PNC_KONFLUX_TOOLING_IMAGE
+    description: Name of the tooling image.
+    type: string
+- op: add
+  path: /spec/params/-
+  value:
+    name: RECIPE_IMAGE
+    description: The image from the build recipe to use
+    type: string
+
+- op: add
+  path: /spec/steps/2
+  value:
+    name: preprocessor
+    image: $(params.PNC_KONFLUX_TOOLING_IMAGE)
+    securityContext:
+      runAsUser: 0
+    computeResources:
+      limits:
+        cpu: 300m
+        memory: 512Mi
+      requests:
+        cpu: 10m
+        memory: 512Mi
+    args:
+      - prepare
+      - --build-tool-version=$(params.BUILD_TOOL_VERSION)
+      - --java-version=$(params.JAVA_VERSION)
+      - --recipe-image=$(params.RECIPE_IMAGE)
+      - --tooling-image=$(params.PNC_KONFLUX_TOOLING_IMAGE)
+      - --type=$(params.BUILD_TOOL)
+      - /var/workdir/source
```

Comment:
This is a very opaque command. It isn'y clear what this actually does to the source especially since it is running with full access to the internet. I don't know how we would be able to qualify/verify our trust in the changes made here.

Reply:
Admittedly this command does not need network access and is similar to what we would need if KONFLUX-4903 was available. It runs java code to preprocess the source to ensure its possible to build it inside a container from https://github.com/project-ncl/konflux-tooling and is only meant to be used by middleware/pnc.

Reply:
If the concern is full internet access we can probably run it within "unshare" but I believe it's a matter of trusting the application, that it rally does what it should?

Reply:
If it doesn't need internet access, I'd restrict it.

Reply:
@mmorhun Are you saying this is your only objection before approval? That tooling image does not currently have unshare installed so we would have to modify the image I suppose.

Reply:
I'll look at installing unshare in the tooling image and running this command with `unshare -n /opt/jboss/container/java/run/run-java.sh...`

Reply:
Unfortunately I get `unshare: unshare failed: Operation not permitted` when trying to run the above command

Reply:
Rather than unshare, would it be acceptable to simply define `-DsocksProxyHost=does.not.exist` which according to https://docs.oracle.com/javase/7/docs/technotes/guides/net/proxies.html should force all outgoing TCP connections to go via the proxy (which will then fail) ?

Reply:
I have not yet added anything to isolate the network. *If* network isolation is mandatory then I could add
```
  +    script: |
  +      #!/bin/bash
  +      /opt/jboss/container/java/run/run-java.sh -DsocksProxyHost=does.not.exist $@
```
to the task to isolate it.

Reply:
I'd defer it to EC team and prodsec

Reply:
Can they be linked in to give their opinion please?

Reply:
I don't have a strong opinion on the security implications. Re the opaqueness concern, maybe a comment with a link to the source code for the command being run would help address that. Re the "it should run without internet access", I agree it's a good idea, but consider filing a separate Jira for tracking that work and implementing it in a future PR, rather than have it block this one. No need to do everything in one PR, right?

Reply:
Re `link to the source code for the command` ; its now linked in the readme

Reply:
From the ProdSec side I don't have any strong opinion.  I will ask some of my team mates if they see any blockers here. My only concern is related to the fact how we can register what was cloned from the provided url? Do we want to verify if the url is trusted somehow? Can we precisely register the source url and manifest the content?

Reply:
@p-rog I'm not entirely sure what you mean; I've adapted the git-clone task as KONFLUX-4903 is not yet available.

Reply:
The purpose of this rebuild steps is to collect the java source code to ensure its possible to build it inside the container. My concerns are related to the sources manifesting and potential chance of losing information about the source code and artifacts used to build the particular component, which is crucial from the later vulnerability management and supply chain security risk assessment perspective. But maybe I misunderstand something here and these concerns are unnecessary.

Reply:
I think this is diverging off network isolation and perhaps should be entered as a JIRA? Or we can discuss in slack with @matejonnet.  If you're talking generating sboms that would be handled by sbomer I guess?

Reply:
> unshare: unshare failed: Operation not permitted

The processing here is just calling a script in your image. Did you try to use unshare within the command in your image that was called?

What are the possible impacts of the user-provided script parameter? Are we sufficiently protecting against maliciously crafted content?

Reply:
The processing here is running a java program and so unshare couldn't be used within it though I did experiment with unshare running the program but it seemed  using `-DsocksProxyHost` would provide the same effect.
The user provided script is not run in this stage but embedded into the build script. It comes from PNC and is used in the build stage (e.g. mvn -D.... install or gradle -P.....assemble).